hustle_public_api_client/samples/upload_leads_csv_with_van_integration.py

```
@@ -0,0 +1,145 @@
+        super().__init__(token='', *args, **kwargs)
+        self.client_id_stored = client_id
+        self.client_secret_stored = client_secret
+        self.refresh_token()
+
+    def refresh_token(self):
+        unauth_client = Client(base_url=base_url)
+        credentials: PostOauthTokenBody = PostOauthTokenBody.from_dict({
+            'client_id': self.client_id_stored,
+            'client_secret': self.client_secret_stored,
+            'grant_type': 'client_credentials',
+        })
+        token = post_oauth_token.sync_detailed(client=unauth_client, body=credentials)
+        if token.status_code != 200:
+            raise Exception('Credentials not valid')
+        self.token = token.parsed.access_token
+
+
+def get_integration_from_api(organization_id: Organization.id) -> Integration.id:
+    request = get_integrations.sync_detailed(client=client, organization_id=organization_id, integration_type='van')
+    if request.status_code == 401:
+        client.refresh_token()
+        request = get_organizations.sync_detailed(client=client)
+    while request.status_code == 429:
+        time.sleep(1000)
+        request = get_organizations.sync_detailed(client=client)
+    if request.status_code != 200:
+        raise Exception('Received unexpected error code on GET Integrations: {} '
+                        'with message {}'.format(request.status_code, request.parsed.message))
+    print(request.parsed)
+    if request.parsed.items.count != 1:
+        raise Exception('Organization does not have van integration')
+    return request.parsed.items[0].id
+
+
+def create_lead(
+        lead: PostLeadsBody,
+        integration_id: AttachLeadToIntegrationOperation.integration_id,
+        external_lead_id: VanExternalLeadSchema.external_object.van_id,
+        client: HustleApiClient
+) -> Lead.id:
+    lead_request = post_leads.sync_detailed(client=client, body=lead)
+    if lead_request.status_code == 401:
```

Could this get into an infinite loop if the `refresh_token` function is failing for some reason?
Here and in the other examples.

---

refresh_token should throw an exception on anything besides a successful retrieval which should dump us out of here.